src/Makefile

```diff
@@ -44,6 +44,14 @@ NVCC_FLAGS ?=
 
 LD_FLAGS ?=
 
+ifeq ($(strip $(DEBUG)),1)
```

Comment:
Not an issue in this PR, but I realized we need to be careful that we don't read `DEBUG_LEGATE` in header files, as those headers may be used in the compilation of another downstream library like cunumeric, where `DEBUG_LEGATE` is not defined. A more robust solution would be to record the status of such variables (that are controlled by compile-time flags like `DEBUG`) in a generated `legate_defines.h` header, like legion and realm do with `legion/realm_defines.h` -- also see #8.

Reply:
Correction, this *is* in an issue in this PR. The `CHECK_CUDA_STREAM` macro was moved here from cunumeric, and cunumeric's `cuda_help.h` includes legate.core's `cuda_help.h`. But when cunumeric is being compiled with `--debug` it doesn't define `DEBUG_LEGATE` but `DEBUG_CUNUMERIC`, meaning the instances of `CHECK_CUDA_STREAM` in cunumeric tasks won't do anything.

Reply:
You're right, and I think that issue is well within this PR, as I moved `CHECK_CUDA` and `CHECK_CUDA_STREAM` to the core, and they change their behavior based on the presence of `DEBUG_LEGATE`. Let me think about a solution.

Reply:
Ok. The logic that adds`DEBUG_LEGATE` is moved to `legate.mk` that is shared by all Legate libraries: https://github.com/nv-legate/legate.core/pull/208/files#diff-12c8b10d84c75f3c0d91d58a3c3b8f55caea9427b83f34723ac9c60f8245042bR89 I believe this fixes the issue. @manopapad What do you think?

Reply:
This works, but note that cunumeric sets `DEBUG` (and thus `DEBUG_LEGATE`) independently of legate.core. We will have to be careful not to have `DEBUG_LEGATE` control things like struct layout.